### ISSUE TITLE
[TF-Lite] densenet.pb at /contrib/lite/g3doc/models.md cannot be loaded

### ISSUE THREAD

**Issue by HwMohanLiu** [tensorflow/tensorflow#19316]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: r1.8
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: 0.11.0
- **GCC/Compiler version (if compiling from source)**: 4.9
- **CUDA/cuDNN version**: no
- **GPU model and memory**: no
- **Exact command to reproduce**: 

### Describe the problem
The densenet.pb provided at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/models.md
cannot be converted by toco command line. It can also not be loaded by GFile API.

### Source code / logs
> bazel-bin/tensorflow/contrib/lite/toco/toco \
--input_file="densenet.pb" \
--output_file="quant_densenet.tflite" \
--input_format=TENSORFLOW_GRAPHDEF \
--output_format=TFLITE \
--inference_input_type=QUANTIZED_UINT8 \
--inference_type=QUANTIZED_UINT8 \
--input_shape=1,224,224,3 \
--input_array="input" \
--output_array="final_dense" \
--allow_custom_ops \
--default_ranges_min=0.0 \
--default_ranges_max=6.0 \
--mean_value=127.5 \
--std_value=1.0 

>[libprotobuf ERROR external/protobuf_archive/src/google/protobuf/text_format.cc:288] Error parsing text-format tensorflow.GraphDef: 1:1: Invalid control characters encountered in text.
>[libprotobuf ERROR external/protobuf_archive/src/google/protobuf/text_format.cc:288] Error parsing text-format tensorflow.GraphDef: 1:2: Interpreting non ascii codepoint 139.
>[libprotobuf ERROR external/protobuf_archive/src/google/protobuf/text_format.cc:288] Error parsing text-format tensorflow.GraphDef: 1:2: Expected identifier, got: ‹
>2018-05-16 15:19:01.232421: F tensorflow/contrib/lite/toco/import_tensorflow.cc:2230] Check failed: ParseFromStringEitherTextOrBinary(input_file_contents, tf_graph.get())
Aborted (core dumped)

if using GFile python API to load the pb file:

```
def load_pb_graph(frozen_graph_file):
    with tf.gfile.GFile(frozen_graph_file, 'rb') as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())
    with tf.Graph().as_default() as graph:
        tf.import_graph_def(graph_def, name="prefix")

    return graph, graph_def
```

>Traceback (most recent call last):
  File "GraphCheckerPB.py", line 24, in <module>
    graph, graph_def = load_pb_graph(in_pb_file)
  File "GraphCheckerPB.py", line 14, in load_pb_graph
    graph_def.ParseFromString(f.read())
google.protobuf.message.DecodeError: Error parsing message

**Comment by jart** [comment]
Try `codecs.getreader("utf-8")` and note questions like these are better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

**Comment by HwMohanLiu** [comment]
@jart I tried to add codecs.getreader("utf-8"), but it does not work. 
`with codecs.getreader("utf-8")(tf.gfile.GFile(frozen_graph_file, 'rb')) as f:`
> UnicodeDecodeError: 'utf8' codec can't decode byte 0x8b in position 1: invalid start byte

Using the latest tf mobile demo to load the model is also failed. 

I opened the pb file by Notpad++. The structure of the content is totally different from that of the model created by tf.graph_utils.convert_variables_to_constants and tf.gfile.FastGFile API. 

The purpose of parsing the pb model is to get the input and output name, and then to convert it to tf-lite quantified version.
